src/Tools/dotnet-monitor/Startup.cs

```diff
@@ -183,22 +184,18 @@ public void ConfigureServices(IServiceCollection services)
         private static void LogBoundAddresses(IFeatureCollection features, AddressListenResults results, ILogger logger)
         {
             IServerAddressesFeature serverAddresses = features.Get<IServerAddressesFeature>();
-            Debug.Assert(serverAddresses.Addresses.Count == results.AddressesCount + results.MetricAddressesCount);
 
             // This logging allows the tool to differentiate which addresses
             // are default address and which are metrics addresses.
-            int urlCount = results.AddressesCount;
-            foreach (string serverAddress in serverAddresses.Addresses)
+
+            foreach (string defaultAddress in results.GetDefaultAddresses(serverAddresses))
             {
```

Comment:
Can you make sure to test this with metrics turned off as well?

Reply:
Seems to be working correctly. No addresses are reported via GetMetricsAddresses.